src/Socket.c

```diff
@@ -490,7 +490,7 @@ SOCKET Socket_getReadySocket(int more_work, int timeout, mutex_type mutex, int*
 
 	FUNC_ENTRY;
 	Thread_lock_mutex(mutex);
-	if (mod_s.nfds == 0 && mod_s.saved.nfds == 0)
+	if (mod_s.nfds == 0 || (mod_s.nfds == 0 && mod_s.saved.nfds == 0))
```

Comment:
I don't think this is right. If mod_s.nfds == 0 then we won't get past the ||. If mod_s.nfds != 0 then both sides of the || will be false. So everything including the || and after is redundant.

Reply:
OK .